editions/test/tiddlers/tests/test-fields-widget.js

```diff
@@ -0,0 +1,91 @@
+/*\
+title: test-field-widget.js
+type: application/javascript
+tags: [[$:/tags/test-spec]]
+
+Tests the html output of the fields widget. Was needed because of https://github.com/Jermolene/TiddlyWiki5/issues/2517
+Field values containing 2 or more $-signs didn't survive a "save - reload" round trip.
+
+Additional tests needed:
+ - $$, $&, $`, $', $1
+
+see fields:
+ - a, g, h, i, j,
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+describe("fields-widget tests for save-reload roundtrip", function() {
+
+	var fields = {
+					a: "asdf",
+					b: "$",
+					c: "$$",
+					cc: "$$$",
+					ccc: "$$$$",
+					d: "$param$",
+					e: "$$param$$",
+					f: "$$param$$$",
+					g: "$&",
+					h: "$`",
+					i: "$'",
+					j: "$1"
+	};
+
+	// Prepare the ontent tiddler
+	$tw.wiki.addTiddler(new $tw.Tiddler({title: "TiddlerOne", text: '--'}, fields));
+
+	// Use different templates
+	$tw.wiki.addTiddler({title: "output-a", text: '<$list filter=TiddlerOne template="$:/core/templates/html-div-tiddler"/>'});
```

Comment:
Why use the list widget here instead of a simple `{{TiddlerOne||$:/core/templates/html-div-tiddler}}`?